IDR/notebook/nbconverted/1.describe_screens.py

```diff
@@ -0,0 +1,284 @@
+
+            # Get phenotype identification accession code
+            try:
+                phenotype_identifier_index = int(
+                    [x["ns"] for x in annotations].index('openmicroscopy.org/mapr/phenotype'))
+                phenotype_identifier = {x[0]: x[1] for x in annotations[phenotype_identifier_index]["values"]}[
+                    "Phenotype Term Accession"]
+            except (ValueError, KeyError):
+                phenotype_identifier = "Not Listed"
+
+            # Build results
+            plate_results.append([
+                screen_id,
+                plate,
+                plate_name,
+                id,
+                cell_line,
+                strain,
+                gene_identifier,
+                phenotype_identifier,
+                stain,
+                stain_target,
+                pixel_size_x,
+                pixel_size_y
+            ])
+
+    # Output results
+    plate_results_df = pd.DataFrame(
+        plate_results,
+        columns=[
+            "screen_id",
+            "plate_id",
+            "plate_name",
+            "image_id",
+            "cell_line",
+            "strain",
+            "gene_identifier",
+            "phenotype_identifier",
+            "stain",
+            "stain_target",
+            "pixel_size_x",
+            "pixel_size_y"
+        ]
+    )
+    return plate_results_df
+
+# Load IDR ids
+data_dir = pathlib.Path(
+    "~/Documents/publicly-available-microscopy-data/IDR/data")
+id_file = pathlib.Path(data_dir, "idr_ids.tsv")
+id_df = pd.read_csv(id_file, sep="\t")
+
+# Create http session
+INDEX_PAGE = "https://idr.openmicroscopy.org/webclient/?experimenter=-1"
+with requests.Session() as session:
+    request = requests.Request('GET', INDEX_PAGE)
+    prepped = session.prepare_request(request)
+    response = session.send(prepped)
+    if response.status_code != 200:
+        response.raise_for_status()
+
+# Extract summary details for all screens
+screen_ids = id_df.query("category=='Screen'").id.tolist()
+screen_details_df = (
+    pd.concat([
+       extract_study_info(session=session, screen_id=x) for x in screen_ids
+    ], axis="rows")
+    .reset_index(drop=True)
+)
+
+output_file = pathlib.Path(data_dir, "screen_details.tsv")
+screen_details_df.to_csv(output_file, index=False, sep="\t")
+
+# Initialize Pool object for threading
+start = time.time()
+available_cores = len(os.sched_getaffinity(0))
+pool = multiprocessing.Pool(processes=available_cores)
+print(f"\nNow processing {len(screen_ids)} screens with {available_cores} cpu cores.\n")
+
+# Pull pertinent details about the screen (plates, wells, channels, cell line, etc.)
+plate_results_dfs = pool.map(describe_screen, screen_ids)
+
+# Terminate pool processes
+pool.close()
+pool.join()
+
+# Combine to create full dataframe
+all_plate_results_df = pd.concat(plate_results_dfs, ignore_index=True)
+
+# TODO: Implement a more effective way to do this
```

Comment:
I'm not sure if I saw these sections in the associated `1.describe_screens.ipynb` file.

If you haven't already, please make sure to `nbconvert` your `.ipynb` files to `.py` files using the following command:

```bash
jupyter nbconvert --to=script --FilesWriter.build_directory=scripts/nbconverted *.ipynb
```

(or some other fancy vscode magic command that I'm not aware of!)

Reply:
New `.py` files were generated from `.ipynb` files using your suggested terminal command.